pyo3-macros-backend/src/pyfunction.rs

```diff
@@ -183,9 +188,50 @@ pub fn parse_name_attribute(attrs: &mut Vec<syn::Attribute>) -> syn::Result<Opti
 pub fn build_py_function(ast: &mut syn::ItemFn, args: PyFunctionAttr) -> syn::Result<TokenStream> {
     let python_name =
         parse_name_attribute(&mut ast.attrs)?.unwrap_or_else(|| ast.sig.ident.unraw());
+
     add_fn_to_module(ast, python_name, args)
 }
 
+/// Parses #[pyo3(from_python_with = "func")]
+pub fn parse_arg_attributes(cap: &mut syn::PatType) -> syn::Result<PyFunctionArgAttr> {
+    let mut from_py_with = None;
+    let mut new_attrs = Vec::new();
+
+    for attr in cap.attrs.iter() {
+        if let Ok(syn::Meta::List(meta_list)) = attr.parse_meta() {
+            if meta_list.path.is_ident("pyo3") {
+                for meta in meta_list.nested.iter() {
+                    if let syn::NestedMeta::Meta(syn::Meta::NameValue(name_value)) = meta {
+                        if name_value.path.is_ident("from_py_with") {
+                            if let syn::Lit::Str(string_literal) = &name_value.lit {
+                                let expr_path = string_literal.parse::<syn::ExprPath>()?;
+                                from_py_with = Some(expr_path);
+                            } else {
+                                return Err(syn::Error::new_spanned(
+                                    &name_value.lit,
+                                    "value should be a string literal",
+                                ));
+                            }
+                        } else {
+                            return Err(syn::Error::new_spanned(
+                                name_value,
+                                "Only from_py_with is supported",
+                            ));
+                        };
+                    }
+                }
+            } else {
+                new_attrs.push(attr.clone())
+            }
+        } else {
+            new_attrs.push(attr.clone())
+        }
+    }
+    cap.attrs.clear();
+    cap.attrs.extend(new_attrs);
```

Comment:
Why do we need to remove `from_py_with`?
If you do so because another function raises an error for that, I think there would be a better way.

Reply:
We have to remove the attributes, because we can't yet specify helper attributes in `#[proc_macro_attribute]`
https://github.com/rust-lang/rust/issues/65823

If we don't remove the attributes, the compiler doesn't recognize it.
```
error: cannot find attribute `pyo3` in this scope
   --> tests/test_pyfunction.rs:124:7
    |
124 |     #[pyo3(from_py_with = "datetime_to_timestamp")] timestamp: i64,
    |       ^^^^
```

The same approach is used for `#[pyo3(get, set)]` attributes inside `#[pyclass]`: https://github.com/PyO3/pyo3/blob/88872eba4fbe895a0728e86505caef8c85a98b07/pyo3-macros-backend/src/pyclass.rs#L184-L211

This solution forces us to make function signature mutable, and it's not very nice, so please let me know if there's another way

Reply:
I guess we could do it in two passes:
1. Parse all attributes from an immutable reference to a TokenStream
2. Obtain a mutable reference and remove all pyo3 attributes

Reply:
I understand, thank you for the explanation!
Then I think it's fine to leave this mutable, but let's use `&mut Vec<_>` instead.